src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,647 @@ const ActorInit En_Rz_InitVars = {
+    func_80BFBA50(this, play);
+    this->animIndex = 9;
+    this->actor.targetMode = 0;
+    this->actor.terminalVelocity = -9.0f;
+    this->actor.gravity = -1.0f;
+
+    switch (EN_RZ_GET_PARAM_F(thisx)) {
+        case 1:
+            if (!(gSaveContext.save.weekEventReg[77] & 4)) {
+                Actor_MarkForDeath(&this->actor);
+                return;
+            }
+            EnRz_ChangeAnim(play, this, 3, ANIMMODE_LOOP, 0.0f);
+            this->actionFunc = func_80BFC674;
+            this->actor.shape.yOffset = -1500.0f;
+            break;
+
+        case 2:
+            this->actor.flags |= ACTOR_FLAG_10;
+            if (gSaveContext.save.weekEventReg[77] & 4) {
+                EnRz_ChangeAnim(play, this, 8, ANIMMODE_LOOP, 0.0f);
+            } else {
+                EnRz_ChangeAnim(play, this, 6, ANIMMODE_LOOP, 0.0f);
+            }
+            this->actionFunc = func_80BFC3F8;
+            this->unk_40C = func_80BFBDA0(this, play);
+            this->actor.uncullZoneForward = 300.0f;
+            break;
+
+        default:
+            this->unk_420 |= 2;
+            if (gSaveContext.save.weekEventReg[77] & 4) {
+                Actor_MarkForDeath(&this->actor);
+                return;
+            }
+            EnRz_ChangeAnim(play, this, 2, ANIMMODE_LOOP, 0.0f);
+            this->actionFunc = func_80BFC8F8;
+            if (!EN_RZ_GET_PARAM_8000(thisx)) {
+                this->actor.textId = 0x291C;
+            } else {
+                this->actor.textId = 0x2920;
+            }
+            break;
+    }
+    if (!EN_RZ_GET_PARAM_8000(thisx)) {
+        this->csAction = 0x226;
+    } else {
+        this->csAction = 0x227;
+    }
+
+    this->unk_428 = 0;
+}
+
+/**
+ * Custom shadow draw function of type ActorShadowFunc.
+ */
+void EnRz_ActorShadowFunc(Actor* thisx, Lights* mapper, PlayState* play) {
+    Vec3f oldPos;
+    EnRz* this = THIS;
+
+    if (this->animIndex == 8) {
+        f32 tempScale = (((27.0f - this->shadowPos.y) + this->actor.world.pos.y) * ((1 / 2.25f) * 0.001f)) + 0.01f;
+        this->actor.scale.x = tempScale;
+    }
+    Math_Vec3f_Copy(&oldPos, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->shadowPos);
+    func_800B4AEC(play, &this->actor, 50.0f);
+
+    if (oldPos.y < this->actor.floorHeight) {
+        this->actor.world.pos.y = this->actor.floorHeight;
+    } else {
+        this->actor.world.pos.y = oldPos.y;
+    }
+
+    ActorShadow_DrawCircle(&this->actor, mapper, play);
+    Math_Vec3f_Copy(&this->actor.world.pos, &oldPos);
+    this->actor.scale.x = 0.01f;
+}
+
+void EnRz_ChangeAnim(PlayState* play, EnRz* this, s16 animIndex, u8 animMode, f32 transitionRate) {
+    f32 endFrame;
+    AnimationHeader** animationPtr;
+
+    if (!EN_RZ_GET_PARAM_8000(&this->actor)) {
+        animationPtr = D_80BFCD20;
+    } else {
+        animationPtr = D_80BFCD3C;
+    }
+
+    if ((animIndex >= ANIMMODE_LOOP) && (animIndex < 9) &&
```

Comment:
```suggestion
    if ((animIndex >= 0) && (animIndex < 9) &&
```
That is not an animmode, so the enum doesn't make sense there.